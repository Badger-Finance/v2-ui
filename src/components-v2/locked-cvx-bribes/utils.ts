import BigNumber from 'bignumber.js';
import { formatWithoutExtraZeros, numberWithCommas, unscale } from '../../mobx/utils/helpers';

export function formatBalance(balance: BigNumber): string {
  return numberWithCommas(formatWithoutExtraZeros(unscale(balance, 18)));
}
