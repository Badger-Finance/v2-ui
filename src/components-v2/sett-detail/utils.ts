import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/setts/sett-charts';
import { DelaySeverity } from '../../mobx/model/setts/sett-rewards';
import { SettDetailMode } from '../../mobx/model/setts/sett-detail';

dayjs.extend(utc);

export const ChartModeTitles = {
	[ChartMode.value]: 'Sett Value',
	[ChartMode.ratio]: 'Token Ratio',
	[ChartMode.accountBalance]: 'Your Holdings',
};

export const SettModeTitles = {
	[SettDetailMode.settInformation]: 'Vault Info',
	[SettDetailMode.userInformation]: 'User Info',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};
