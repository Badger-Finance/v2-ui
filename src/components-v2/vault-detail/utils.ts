import { Vault, VaultData } from '@badger-dao/sdk';
import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/vaults/vault-charts';
import { DelaySeverity } from '../../mobx/model/vaults/vault-rewards';

dayjs.extend(utc);

export const ChartModeTitles: Record<string, string> = {
	[ChartMode.Value]: 'Sett Value',
	[ChartMode.Ratio]: 'Token Ratio',
	[ChartMode.AccountBalance]: 'My Holdings',
	[ChartMode.BoostMultiplier]: 'Badger Boost',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};

export function defaultVaultBalance(vault: Vault): VaultData {
	return {
		address: vault.vaultToken,
		name: vault.name,
		symbol: vault.asset,
		pricePerFullShare: vault.pricePerFullShare,
		balance: 0,
		value: 0,
		earnedBalance: 0,
		earnedValue: 0,
		depositedBalance: 0,
		withdrawnBalance: 0,
		tokens: [],
		earnedTokens: [],
	};
}

export interface Balance {
	balance: number;
}

export function hasBalance(input?: Balance): boolean {
	if (!input) {
		return false;
	}
	return input.balance > 0;
}
