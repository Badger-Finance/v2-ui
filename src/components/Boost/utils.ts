import { makeStyles } from '@material-ui/core/styles';
import { ClassNameMap } from '@material-ui/core/styles/withStyles';
import React from 'react';

type ComparisonConfig = {
  toCompareValue: number;
  toBeComparedValue: number;
  greaterCaseColor: string;
  lessCaseColor: string;
  defaultColor: string;
};

export const getColorFromComparison = ({
  toCompareValue,
  toBeComparedValue,
  greaterCaseColor,
  lessCaseColor,
  defaultColor,
}: ComparisonConfig): string => {
  if (toCompareValue > toBeComparedValue) return greaterCaseColor;
  if (toCompareValue < toBeComparedValue) return lessCaseColor;

  return defaultColor;
};

export const useAssetInputStyles = (
  currentValue: string,
  toCompare = 0,
): ((props?: React.CSSProperties) => ClassNameMap<'assetColor'>) => {
  return makeStyles((theme) => {
    const defaultColor = currentValue ? theme.palette.text.primary : theme.palette.text.secondary;
    const fontColor = getColorFromComparison({
      toCompareValue: Number(currentValue),
      toBeComparedValue: toCompare,
      greaterCaseColor: '#74D189',
      lessCaseColor: theme.palette.error.main,
      defaultColor,
    });

    return {
      assetColor: {
        color: fontColor,
      },
    };
  });
};
