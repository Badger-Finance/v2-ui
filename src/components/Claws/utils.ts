import BigNumber from 'bignumber.js';
import { INVALID_REASON } from './index';

export interface ValidationParams {
	amount: string | BigNumber;
	maximum: BigNumber;
	minimum?: BigNumber;
}

export function validateAmountBoundaries({
	amount,
	maximum,
	minimum = new BigNumber('0'),
}: ValidationParams): INVALID_REASON | undefined {
	let error: INVALID_REASON | undefined = undefined;

	const input = new BigNumber(amount);
	const amountExceedsBalance = input.gt(maximum);
	const amountIsLessThanMinimum = input.lt(minimum);

	if (amountIsLessThanMinimum) error = INVALID_REASON.UNDER_MINIMUM;
	if (amountExceedsBalance) error = INVALID_REASON.OVER_MAXIMUM;

	console.log({
		input: input.toString(),
		minimum: minimum.toString(),
		maximum: maximum.toString(),
		amountExceedsBalance,
		amountIsLessThanMinimum,
		error,
	});

	return error;
}
