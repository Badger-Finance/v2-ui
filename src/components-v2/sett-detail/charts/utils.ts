import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { Sett } from '../../../mobx/model/setts/sett';
import { fetchSettChartInformation } from '../../../mobx/utils/apiV2';
import { SettSnapshotGranularity } from '../../../mobx/model/setts/sett-snapshot';
import { SettChartData } from '../../../mobx/model/setts/sett-charts';

dayjs.extend(utc);

export enum SettChartTimeframe {
	'day' = 1,
	'week' = 7,
	'month' = 30,
}

/**
 * Fetches chart information since the provided timeframe until the current date
 * @param sett
 * @param timeframe
 */
export const fetchSettChart = async (sett: Sett, timeframe: SettChartTimeframe): Promise<SettChartData[] | null> => {
	const timeframeDays = 1 * timeframe;
	const isDayTimeFrame = timeframe === SettChartTimeframe.day;

	// if timeframe is just one day then we want the granularity to be hours
	const granularity = isDayTimeFrame ? SettSnapshotGranularity.HOUR : SettSnapshotGranularity.DAY;

	const now = dayjs().utc(); // query until current date
	const from = dayjs(now).subtract(timeframeDays, 'days').utc();

	const fetchedData = await fetchSettChartInformation(sett.vaultToken, from.toDate(), now.toDate(), granularity);

	if (!fetchedData) {
		return null;
	}

	// data needs to be ascending sorted
	return fetchedData
		.sort((a, b) => a.timestamp - b.timestamp)
		.map((d) => ({ ...d, timestamp: new Date(d.timestamp) }));
};
