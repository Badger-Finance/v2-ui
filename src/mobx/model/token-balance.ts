import BigNumber from 'bignumber.js';
import RewardsStore from 'mobx/stores/rewardsStore';
import { inCurrency, minBalance } from 'mobx/utils/helpers';
import { BadgerToken } from './badger-token';

export class TokenBalance {
	private store: RewardsStore;
	readonly token: BadgerToken;
	public tokenBalance: BigNumber;
	public balance: BigNumber;
	public price: BigNumber;

	constructor(store: RewardsStore, token: BadgerToken, balance: BigNumber, price: BigNumber) {
		this.store = store;
		this.token = token;
		this.tokenBalance = balance;
		this.price = price;
		this.balance = balance.dividedBy(Math.pow(10, token.decimals));
	}

	/**
	 * Create a token balance from a string balance of appropriate decimals.
	 * i.e.
	 *   Given token A as a defined TokenBalance.
	 *   TokenBalance.fromBalance(A, A.balanceDisplay()) = A;
	 * Above does not hold true given a balance display of balance below a
	 * a requested precision threshold (< 0.001 balance display).
	 * Does not work with digg share conversion - only fragments support.
	 */
	static fromBalance(tokenBalance: TokenBalance, balance: string): TokenBalance {
		const { token, store, price } = tokenBalance;
		const scalar = new BigNumber(Math.pow(10, token.decimals));
		const amount = new BigNumber(balance).multipliedBy(scalar);
		return new TokenBalance(store, token, amount, price);
	}

	get value(): BigNumber {
		return this.balance.multipliedBy(this.price);
	}

	/**
	 * Return a string display value for a given precision. Defaults to token decimals.
	 * i.e.
	 *  0 => 0.0 (1 decimal)
	 *  0.0001 => < 0.001 (3 decimals)
	 *  0.0001 => 0.00010 (5 decimals)
	 * @param precision decimal count for display purposes
	 * @returns string representation of balance
	 */
	balanceDisplay(precision?: number): string {
		const decimals = precision || this.token.decimals;
		if (this.balance.gt(0) && this.balance.lt(minBalance(decimals))) {
			return `< 0.${'0'.repeat(decimals - 1)}1`;
		}
		return this.balance.toFixed(decimals);
	}

	balanceValueDisplay(currency: string): string {
		return inCurrency(this.value, currency);
	}

	scale(scalar: BigNumber): TokenBalance {
		const tokenBalance = this.tokenBalance.multipliedBy(scalar);
		return new TokenBalance(this.store, this.token, tokenBalance, this.price);
	}

	scaledBalanceDisplay(percent: number): string {
		const scaledBalance = this.scale(new BigNumber(percent / 100));
		return scaledBalance.balanceDisplay();
	}
}
