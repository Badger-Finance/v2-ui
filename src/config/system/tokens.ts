import { ERC20, BSC_ERC20, NETWORK_LIST, NETWORK_CONSTANTS } from 'config/constants';
import deploy from '../deployments/mainnet.json';
import bscDeploy from '../deployments/bsc.json';
import { AbiItem } from 'web3-utils';
import { TokenNetworkConfig } from 'mobx/model';
import {zipObject} from "../../utils/lodashToNative";

export const getTokens = (network?: string | null): TokenNetworkConfig | undefined => {
	switch (network) {
		case NETWORK_LIST.BSC:
			const bscVaultList = [
				bscDeploy.sett_system.vaults['native.pancakeBnbBtcb'],
				bscDeploy.sett_system.vaults['native.bBadgerBtcb'],
				bscDeploy.sett_system.vaults['native.bDiggBtcb'],
			];
			const bscTokenList = [
				'0x7561EEe90e24F3b348E1087A005F78B4c8453524',
				'0x10F461CEAC7A17F59e249954Db0784d42EfF5DB5',
				'0xE1E33459505bB3763843a426F7Fd9933418184ae',
			];
			const bscTokenMap = zipObject(bscVaultList, bscTokenList);
			return {
				priceEndpoints: ['https://api.thegraph.com/subgraphs/name/pancakeswap/exchange'],
				tokenBatches: [
					{
						abi: BSC_ERC20.abi as AbiItem[],
						methods: [
							{
								name: 'totalSupply',
							},
							{
								name: 'balanceOf',
								args: ['{connectedAddress}'],
							},
						],
						contracts: [
							'0x753fbc5800a8C8e3Fb6DC6415810d627A387Dfc9', // BADGER
							'0x1F7216fdB338247512Ec99715587bb97BBf96eae', // bBADGER
							'0x5986D5c77c65e5801a5cAa4fAE80089f870A71dA', // bDigg
							'0x7561EEe90e24F3b348E1087A005F78B4c8453524', // bnbBtc Cake LP
							'0x7130d2A12B9BCbFAe4f2634d864A1Ee1Ce3Ead9c', // BTCb
							'0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c', // BNB
							'0x0E09FaBB73Bd3Ade0a17ECC321fD13a19e81cE82', // CAKE
							'0x10F461CEAC7A17F59e249954Db0784d42EfF5DB5', // bBadger/BTCb Cake LP
							'0xE1E33459505bB3763843a426F7Fd9933418184ae', // bDIGG/BTCb Cake LP
						],
					},
				],
				decimals: {
					'0x753fbc5800a8C8e3Fb6DC6415810d627A387Dfc9': 18, // BADGER
					'0x1F7216fdB338247512Ec99715587bb97BBf96eae': 18, // bBADGER
					'0x5986D5c77c65e5801a5cAa4fAE80089f870A71dA': 18, // bDigg
					'0x7561EEe90e24F3b348E1087A005F78B4c8453524': 18, // bnbBtc Cake LP
					'0x7130d2A12B9BCbFAe4f2634d864A1Ee1Ce3Ead9c': 18, // BTCb
					'0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c': 18, // BNB
					'0x0E09FaBB73Bd3Ade0a17ECC321fD13a19e81cE82': 18, // CAKE
					'0x10F461CEAC7A17F59e249954Db0784d42EfF5DB5': 18, // bBadger/BTCb Cake LP
					'0xE1E33459505bB3763843a426F7Fd9933418184ae': 18, // bDIGG/BTCb Cake LP
				},
				symbols: {
					'0x753fbc5800a8C8e3Fb6DC6415810d627A387Dfc9': 'BADGER', // BADGER
					'0x1F7216fdB338247512Ec99715587bb97BBf96eae': 'bBADGER', // bBADGER
					'0x5986D5c77c65e5801a5cAa4fAE80089f870A71dA': 'bDigg', // bDigg
					'0x7561EEe90e24F3b348E1087A005F78B4c8453524': 'BNB/BTCb CAKE-LP', // bnbBtc Cake LP
					'0x7130d2A12B9BCbFAe4f2634d864A1Ee1Ce3Ead9c': 'BTCb', // BTCb
					'0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c': 'BNB', // BNB
					'0x0E09FaBB73Bd3Ade0a17ECC321fD13a19e81cE82': 'CAKE', // CAKE
					'0x10F461CEAC7A17F59e249954Db0784d42EfF5DB5': 'bBADGER/BTCb CAKE-LP', // bBadger/BTCb Cake LP
					'0xE1E33459505bB3763843a426F7Fd9933418184ae': 'bDIGG/BTCb CAKE-LP', // bDIGG/BTCb Cake LP
				},
				names: {
					'0x7561EEe90e24F3b348E1087A005F78B4c8453524': 'BNB/BTCb Pancake LP',
					'0x753fbc5800a8C8e3Fb6DC6415810d627A387Dfc9': 'Badger',
					'0x1F7216fdB338247512Ec99715587bb97BBf96eae': 'bBadger',
					'0x5986D5c77c65e5801a5cAa4fAE80089f870A71dA': 'bDigg',
					'0x7130d2A12B9BCbFAe4f2634d864A1Ee1Ce3Ead9c': 'BTCb',
					'0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c': 'BNB',
					'0x0E09FaBB73Bd3Ade0a17ECC321fD13a19e81cE82': 'Pancakeswap Token',
					'0x10F461CEAC7A17F59e249954Db0784d42EfF5DB5': 'bBadger/BTCb Pancake LP', // bBadger/BTCb Cake LP
					'0xE1E33459505bB3763843a426F7Fd9933418184ae': 'bBadger/BTCb Pancake LP', // bDIGG/BTCb Cake LP
				},
				tokenMap: bscTokenMap,
			};
		case NETWORK_LIST.ETH:
			const vaultList = [
				deploy.sett_system.vaults['native.sushiDiggWbtc'],
				deploy.sett_system.vaults['native.sbtcCrv'],
				deploy.sett_system.vaults['native.digg'],
				deploy.sett_system.vaults['native.badger'],
				deploy.sett_system.vaults['harvest.renCrv'],
				deploy.sett_system.vaults['native.renCrv'],
				deploy.sett_system.vaults['native.sushiBadgerWbtc'],
				deploy.sett_system.vaults['native.uniBadgerWbtc'],
				deploy.sett_system.vaults['native.sushiWbtcEth'],
				deploy.sett_system.vaults['native.uniDiggWbtc'],
				deploy.sett_system.vaults['native.tbtcCrv'],
			];
			const tokenList = [
				'0x9a13867048e01c663ce8Ce2fE0cDAE69Ff9F35E3',
				'0x075b1bb99792c9E1041bA13afEf80C91a1e70fB3',
				'0x798D1bE841a82a273720CE31c822C61a67a601C3',
				'0x3472A5A71965499acd81997a54BBA8D852C6E53d',
				'0x49849C98ae39Fff122806C06791Fa73784FB3675',
				'0x49849C98ae39Fff122806C06791Fa73784FB3675',
				'0x110492b31c59716AC47337E616804E3E3AdC0b4a',
				'0xcD7989894bc033581532D2cd88Da5db0A4b12859',
				'0xCEfF51756c56CeFFCA006cD410B03FFC46dd3a58',
				'0xE86204c4eDDd2f70eE00EAd6805f917671F56c52',
				'0x64eda51d3Ad40D56b9dFc5554E06F94e1Dd786Fd',
			];
			const tokenMap = zipObject(vaultList, tokenList);
			return {
				curveTokens: {
					contracts: [
						'0x49849C98ae39Fff122806C06791Fa73784FB3675', //renBTC/wBTC (crvRenWBTC)
						'0x64eda51d3Ad40D56b9dFc5554E06F94e1Dd786Fd', //Curve.fi tBTC/sbtcCrv (tbtc/sbtc...)
						'0x075b1bb99792c9E1041bA13afEf80C91a1e70fB3', //Curve.fi renBTC/wBTC/sBTC
					],
					priceEndpoints: [
						'https://stats.curve.fi/raw-stats/ren2-1440m.json',
						'https://stats.curve.fi/raw-stats/tbtc-1440m.json',
						'https://stats.curve.fi/raw-stats/rens-1440m.json',
					],
					names: ['Curve.fi renBTC/wBTC', 'Curve.fi tBTC/sbtcCrv', 'Curve.fi renBTC/wBTC/sBTC'],
					vsToken: NETWORK_CONSTANTS[NETWORK_LIST.ETH].TOKENS.WBTC_ADDRESS,
				},
				priceEndpoints: [
					'https://api.thegraph.com/subgraphs/name/jiro-ono/sushiswap-v1-exchange',
					'https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v2',
				],
				tokenBatches: [
					{
						abi: ERC20.abi as AbiItem[],
						methods: [
							{
								name: 'totalSupply',
							},
							{
								name: 'balanceOf',
								args: ['{connectedAddress}'],
							},
						],
						contracts: [
							'0x6B3595068778DD592e39A122f4f5a5cF09C90fE2',
							'0x6dEf55d2e18486B9dDfaA075bc4e4EE0B28c1545',
							'0x9a13867048e01c663ce8Ce2fE0cDAE69Ff9F35E3',
							'0x19D97D8fA813EE2f51aD4B4e04EA08bAf4DFfC28',
							'0x36e2FCCCc59e5747Ff63a03ea2e5C0c2C14911e7',
							'0x64eda51d3Ad40D56b9dFc5554E06F94e1Dd786Fd',
							'0x075b1bb99792c9E1041bA13afEf80C91a1e70fB3',
							'0x235c9e24D3FB2FAFd58a2E49D454Fdcd2DBf7FF1',
							'0x758A43EE2BFf8230eeb784879CdcFF4828F2544D',
							'0x798D1bE841a82a273720CE31c822C61a67a601C3',
							'0x1862A18181346EBd9EdAf800804f89190DeF24a5',
							'0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599',
							'0x3472A5A71965499acd81997a54BBA8D852C6E53d',
							'0x49849C98ae39Fff122806C06791Fa73784FB3675',
							'0x110492b31c59716AC47337E616804E3E3AdC0b4a',
							'0xAf5A1DECfa95BAF63E0084a35c62592B774A2A87',
							'0xb9D076fDe463dbc9f915E5392F807315Bf940334',
							'0xcD7989894bc033581532D2cd88Da5db0A4b12859',
							'0xCEfF51756c56CeFFCA006cD410B03FFC46dd3a58',
							'0xd04c48A53c111300aD41190D63681ed3dAd998eC',
							'0xE86204c4eDDd2f70eE00EAd6805f917671F56c52',
							'0x88128580ACdD9c04Ce47AFcE196875747bF2A9f6',
							'0x8798249c2E607446EfB7Ad49eC89dD1865Ff4272',
						],
					},
				],
				decimals: {
					'0x6B3595068778DD592e39A122f4f5a5cF09C90fE2': 18,
					'0x6dEf55d2e18486B9dDfaA075bc4e4EE0B28c1545': 18,
					'0x9a13867048e01c663ce8Ce2fE0cDAE69Ff9F35E3': 18,
					'0x19D97D8fA813EE2f51aD4B4e04EA08bAf4DFfC28': 18,
					'0x36e2FCCCc59e5747Ff63a03ea2e5C0c2C14911e7': 18,
					'0x64eda51d3Ad40D56b9dFc5554E06F94e1Dd786Fd': 18,
					'0x075b1bb99792c9E1041bA13afEf80C91a1e70fB3': 18,
					'0x235c9e24D3FB2FAFd58a2E49D454Fdcd2DBf7FF1': 18,
					'0x758A43EE2BFf8230eeb784879CdcFF4828F2544D': 18,
					'0x798D1bE841a82a273720CE31c822C61a67a601C3': 9,
					'0x1862A18181346EBd9EdAf800804f89190DeF24a5': 18,
					'0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599': 8,
					'0x3472A5A71965499acd81997a54BBA8D852C6E53d': 18,
					'0x49849C98ae39Fff122806C06791Fa73784FB3675': 18,
					'0x110492b31c59716AC47337E616804E3E3AdC0b4a': 18,
					'0xAf5A1DECfa95BAF63E0084a35c62592B774A2A87': 18,
					'0xb9D076fDe463dbc9f915E5392F807315Bf940334': 18,
					'0xcD7989894bc033581532D2cd88Da5db0A4b12859': 18,
					'0xCEfF51756c56CeFFCA006cD410B03FFC46dd3a58': 18,
					'0xd04c48A53c111300aD41190D63681ed3dAd998eC': 18,
					'0xE86204c4eDDd2f70eE00EAd6805f917671F56c52': 18,
					'0x88128580ACdD9c04Ce47AFcE196875747bF2A9f6': 18,
					'0x8798249c2E607446EfB7Ad49eC89dD1865Ff4272': 18,
				},
				symbols: {
					'0x6B3595068778DD592e39A122f4f5a5cF09C90fE2': 'SUSHI',
					'0x9a13867048e01c663ce8Ce2fE0cDAE69Ff9F35E3': 'wbtcDiggSLP',
					'0x36e2FCCCc59e5747Ff63a03ea2e5C0c2C14911e7': 'xSushiWethSLP',
					'0x64eda51d3Ad40D56b9dFc5554E06F94e1Dd786Fd': 'tbtc/sbtcCrv',
					'0x075b1bb99792c9E1041bA13afEf80C91a1e70fB3': 'crvRenWSBTC',
					'0x798D1bE841a82a273720CE31c822C61a67a601C3': 'DIGG',
					'0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599': 'WBTC',
					'0x3472A5A71965499acd81997a54BBA8D852C6E53d': 'BADGER',
					'0x49849C98ae39Fff122806C06791Fa73784FB3675': 'crvRenWBTC',
					'0x110492b31c59716AC47337E616804E3E3AdC0b4a': 'wbtcBadgerSLP',
					'0xcD7989894bc033581532D2cd88Da5db0A4b12859': 'wbtcBadgerUNI-V2',
					'0xCEfF51756c56CeFFCA006cD410B03FFC46dd3a58': 'wbtcWethSLP',
					'0xE86204c4eDDd2f70eE00EAd6805f917671F56c52': 'wbtcDiggUNI-V2',
					'0x8798249c2E607446EfB7Ad49eC89dD1865Ff4272': 'xSUSHI',

					'0x6dEf55d2e18486B9dDfaA075bc4e4EE0B28c1545': 'bcrvRenWBTC',
					'0x19D97D8fA813EE2f51aD4B4e04EA08bAf4DFfC28': 'bBADGER',
					'0x235c9e24D3FB2FAFd58a2E49D454Fdcd2DBf7FF1': 'bUNI-V2',
					'0x758A43EE2BFf8230eeb784879CdcFF4828F2544D': 'bSLP',
					'0x1862A18181346EBd9EdAf800804f89190DeF24a5': 'bSLP',
					'0x88128580ACdD9c04Ce47AFcE196875747bF2A9f6': 'bSLP',
					'0xAf5A1DECfa95BAF63E0084a35c62592B774A2A87': 'bSupercrvRenWBTC',
					'0xb9D076fDe463dbc9f915E5392F807315Bf940334': 'btbtc/sbtcCrv',
					'0xd04c48A53c111300aD41190D63681ed3dAd998eC': 'bcrvRenWSBTC',
				},
				names: {
					'0x6B3595068778DD592e39A122f4f5a5cF09C90fE2': 'SUSHI',
					'0x9a13867048e01c663ce8Ce2fE0cDAE69Ff9F35E3': 'wBTC/DIGG Sushi LP',
					'0xE86204c4eDDd2f70eE00EAd6805f917671F56c52': 'wBTC/DIGG Uni LP',
					'0x36e2FCCCc59e5747Ff63a03ea2e5C0c2C14911e7': 'xSUSHI/wETH Sushi LP',
					'0x64eda51d3Ad40D56b9dFc5554E06F94e1Dd786Fd': 'Curve.fi tBTC/sBTCCrv LP',
					'0x075b1bb99792c9E1041bA13afEf80C91a1e70fB3': 'Curve.fi crvRenWSBTC',
					'0x798D1bE841a82a273720CE31c822C61a67a601C3': 'Digg',
					'0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599': 'Wrapped Bitcoin',
					'0x3472A5A71965499acd81997a54BBA8D852C6E53d': 'Badger',
					'0x49849C98ae39Fff122806C06791Fa73784FB3675': 'Curve.fi crvRenWBTC',
					'0x110492b31c59716AC47337E616804E3E3AdC0b4a': 'wBTC/Badger Sushi LP',
					'0xcD7989894bc033581532D2cd88Da5db0A4b12859': 'wBTC/Badger Uni LP',
					'0xCEfF51756c56CeFFCA006cD410B03FFC46dd3a58': 'wBTC/wETH Sushi LP',
					'0x8798249c2E607446EfB7Ad49eC89dD1865Ff4272': 'xSUSHI',

					'0x6dEf55d2e18486B9dDfaA075bc4e4EE0B28c1545': 'bcrvRenWBTC',
					'0x19D97D8fA813EE2f51aD4B4e04EA08bAf4DFfC28': 'bBADGER',
					'0x235c9e24D3FB2FAFd58a2E49D454Fdcd2DBf7FF1': 'bUNI-V2',
					'0x758A43EE2BFf8230eeb784879CdcFF4828F2544D': 'bSLP',
					'0x1862A18181346EBd9EdAf800804f89190DeF24a5': 'bSLP',
					'0x88128580ACdD9c04Ce47AFcE196875747bF2A9f6': 'bSLP',
					'0xAf5A1DECfa95BAF63E0084a35c62592B774A2A87': 'bSupercrvRenWBTC',
					'0xb9D076fDe463dbc9f915E5392F807315Bf940334': 'btbtc/sbtcCrv',
					'0xd04c48A53c111300aD41190D63681ed3dAd998eC': 'bcrvRenWSBTC',
				},
				tokenMap: tokenMap,
			};
		default:
			return undefined;
	}
};
