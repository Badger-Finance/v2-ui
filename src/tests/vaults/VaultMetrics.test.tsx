import BigNumber from 'bignumber.js';
import React from 'react';

import VaultMetrics from '../../components-v2/vault-detail/specs/VaultMetrics';
import { TokenBalance } from '../../mobx/model/tokens/token-balance';
import LockedDepositsStore from '../../mobx/stores/LockedDepositsStore';
import { SAMPLE_VAULT } from '../utils/samples';
import { checkSnapshot } from '../utils/snapshots';

describe('VaultMetrics', () => {
	it('renders correctly', () => {
		checkSnapshot(<VaultMetrics vault={SAMPLE_VAULT} />);
	});

	it('displays withdrawable tokens', () => {
		jest.spyOn(LockedDepositsStore.prototype, 'getLockedDepositBalances').mockReturnValue(
			new TokenBalance(
				{
					address: SAMPLE_VAULT.vaultToken,
					symbol: SAMPLE_VAULT.vaultAsset,
					name: SAMPLE_VAULT.name,
					decimals: 18,
				},
				new BigNumber(100 * 1e18),
				new BigNumber(0),
			),
		);
		checkSnapshot(<VaultMetrics vault={SAMPLE_VAULT} />);
	});
});
