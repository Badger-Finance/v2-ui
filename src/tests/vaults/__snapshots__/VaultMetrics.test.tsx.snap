// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaultMetrics displays withdrawable tokens 1`] = `
<div>
  <div
    class="MuiGrid-root-10 makeStyles-root-1 MuiGrid-container-11"
  >
    <h6
      class="MuiTypography-root-113 makeStyles-title-7 MuiTypography-h6-123"
    >
      Vault Details
    </h6>
    <hr
      class="MuiDivider-root-145 WithStyles(ForwardRef(Divider))-root-143 WithStyles(ForwardRef(Divider))-root-144"
    />
    <p
      class="MuiTypography-root-113 makeStyles-amount-2 MuiTypography-body1-115"
    >
      135,697,015
    </p>
    <p
      class="MuiTypography-root-113 MuiTypography-body2-114"
    >
      Assets Deposited
    </p>
    <div
      class="makeStyles-showMoreContainer-8"
    >
      <div
        class="makeStyles-showMore-9"
      >
        Hide
      </div>
    </div>
    <div
      class="MuiCollapse-root-152 MuiCollapse-entered-153"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper-155"
      >
        <div
          class="MuiCollapse-wrapperInner-156"
        >
          <div
            class="makeStyles-submetric-4"
          >
            <p
              class="MuiTypography-root-113 makeStyles-submetricValue-5 MuiTypography-body1-115"
            >
              1.0082
            </p>
            <span
              class="MuiTypography-root-113 makeStyles-submetricType-6 MuiTypography-caption-116"
            >
              tokens per share
            </span>
          </div>
          <div
            class="makeStyles-submetric-4"
          >
            <p
              class="MuiTypography-root-113 makeStyles-submetricValue-5 MuiTypography-body1-115"
            >
              100.00000
            </p>
            <span
              class="MuiTypography-root-113 makeStyles-submetricType-6 MuiTypography-caption-116"
            >
              tokens withdrawable
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`VaultMetrics renders correctly 1`] = `
<div>
  <div
    class="MuiGrid-root-10 makeStyles-root-1 MuiGrid-container-11"
  >
    <h6
      class="MuiTypography-root-113 makeStyles-title-7 MuiTypography-h6-123"
    >
      Vault Details
    </h6>
    <hr
      class="MuiDivider-root-145 WithStyles(ForwardRef(Divider))-root-143 WithStyles(ForwardRef(Divider))-root-144"
    />
    <p
      class="MuiTypography-root-113 makeStyles-amount-2 MuiTypography-body1-115"
    >
      135,697,015
    </p>
    <p
      class="MuiTypography-root-113 MuiTypography-body2-114"
    >
      Assets Deposited
    </p>
    <div
      class="makeStyles-showMoreContainer-8"
    >
      <div
        class="makeStyles-showMore-9"
      >
        Hide
      </div>
    </div>
    <div
      class="MuiCollapse-root-152 MuiCollapse-entered-153"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper-155"
      >
        <div
          class="MuiCollapse-wrapperInner-156"
        >
          <div
            class="makeStyles-submetric-4"
          >
            <p
              class="MuiTypography-root-113 makeStyles-submetricValue-5 MuiTypography-body1-115"
            >
              1.0082
            </p>
            <span
              class="MuiTypography-root-113 makeStyles-submetricType-6 MuiTypography-caption-116"
            >
              tokens per share
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
